### PR TITLE
VoIP: DTMF Support

### PR DESCRIPTION
**Pull request by ismailgulek**
Part of vector-im/element-ios#3929

### CODE REVIEW

**Review by dbkr — APPROVED** [review]

File: MatrixSDK/VoIP/CallStack/MXCallStackCall.h [hunk 1]
```diff
@@ -140,6 +140,40 @@ NS_ASSUME_NONNULL_BEGIN
...
+#pragma mark - DTMF
+
+/**
+ Indicates whether this call can send DTMF tones.
+ This property will be false if the call is not connected yet.
+ */
+@property (nonatomic, readonly) BOOL canSendDTMF;
+
+/**
+ * Queues a task that sends the DTMF tones. The tones parameter is treated
```
Comment (manuroe) [inline MatrixSDK/VoIP/CallStack/MXCallStackCall.h:152]:
```suggestion
 Queues a task that sends the DTMF tones. The tones parameter is treated
```

and etc

Comment (ismailgulek) [reply MatrixSDK/VoIP/CallStack/MXCallStackCall.h]: Comments rewritten.

File: MatrixSDK/VoIP/CallStack/MXCallStackCall.h [hunk 2]
```diff
@@ -140,6 +140,28 @@ NS_ASSUME_NONNULL_BEGIN
...
 
+#pragma mark - DTMF
+
+/**
+ Indicates whether this call can send DTMF tones.
+ This property will be false if the call is not connected yet.
+ */
+@property (nonatomic, readonly) BOOL canSendDTMF;
+
+/// Creates a task to send given DTMF tones in the call. If there is a task already running, it'll be canceled.
```
Comment (manuroe) [inline MatrixSDK/VoIP/CallStack/MXCallStackCall.h:151]: Can we use the same comment style as the existing one is this file?

File: MatrixSDK/VoIP/MXCall.h [hunk 3]
```diff
@@ -184,6 +184,29 @@ extern NSString *const kMXCallSupportsTransferringStatusDidChange;
...
+/// Allowed interval is from 70 ms to 6000 ms inclusively.
+/// If given value is outside of these limits, it'll be limited to them.
+/// Pass 0 to use default value or last used value.
+/// @param interToneGap Duration for gap between each character of tones (in milliseconds).
+/// Must be at least 50 ms.
+/// If given value is lower than 50 ms, it'll be limited to that value.
+/// Pass 0 to use default value or last used value.
+/// @returns Whether the operation succeeded or not.
+- (BOOL)sendDTMF:(NSString * _Nonnull)tones
+        duration:(NSTimeInterval)duration
```
Comment (manuroe) [inline MatrixSDK/VoIP/MXCall.h:207]: Can we use NSUInteger to express milliseconds?
Apple says: *A NSTimeInterval value is always specified in seconds*

File: MatrixSDK/VoIP/MXCall.h [hunk 4]
```diff
@@ -184,6 +184,29 @@ extern NSString *const kMXCallSupportsTransferringStatusDidChange;
...
+#pragma mark - DTMF
+
+/**
+ Indicates whether this call can send DTMF tones.
+ This property will be false if the call is not connected yet.
+ */
+@property (nonatomic, readonly) BOOL supportsDTMF;
+
+/// Creates a task to send given DTMF tones in the call. If there is a task already running, it'll be canceled.
+/// @param tones DTMF tones to be sent. Allowed characters: [0-9], [A-D], '#', `*`. Case insensitive. Comma (',') will cause a 2 seconds delay before sending next character.
```
Comment (manuroe) [inline MatrixSDK/VoIP/MXCall.h:196]: comment style

File: MatrixSDK/VoIP/CallStack/MXCallStackCall.h (hunk 2 above)
Comment (ismailgulek) [reply MatrixSDK/VoIP/CallStack/MXCallStackCall.h]: 👍

File: MatrixSDK/VoIP/MXCall.h (hunk 3 above)
Comment (ismailgulek) [reply MatrixSDK/VoIP/MXCall.h]: 👍

File: MatrixSDK/VoIP/MXCall.h (hunk 4 above)
Comment (ismailgulek) [reply MatrixSDK/VoIP/MXCall.h]: 👍

**Review by manuroe — APPROVED** [review]

**Pull request merged**